tests/unit/test_task_utils.py

```diff
@@ -242,3 +243,97 @@ def test_adls_bulk_upload(mock_upload):
 
     adls_bulk_upload.run(file_names=file_names, adls_file_path="any/at/random")
     mock_upload.assert_called_once()
+
+
+def test_anonymize_df_all():
+    data = pd.DataFrame(
+        {
+            "date_col": ["2005-01-01", "2010-01-02", "2023-01-03"],
+            "email": [
+                "john.malkovich@bing.com",
+                "hannah.montana@disney.com",
+                "ilovedogs33@dyvenia.com",
+            ],
+            "name": ["John", "Hannah", "Patryk"],
+            "last_name": ["Malkovich", "Montana", "Dyvenian"],
+            "active": [1, 0, 0],
+        }
+    )
+
+    expected_output = {
+        "date_col": {0: "2005-01-01", 1: "2010-01-02", 2: "2023-01-03"},
+        "email": {0: "***", 1: "***", 2: "***"},
+        "name": {0: "***", 1: "***", 2: "***"},
+        "last_name": {0: "***", 1: "***", 2: "***"},
+        "active": {0: 1, 1: 0, 2: 0},
+    }
+    output = anonymize_df.run(data, ["email", "name", "last_name"]).to_dict()
+    assert output == expected_output
+
+
+def test_anonymize_df_hash_defined_threshold_date():
+    data = pd.DataFrame(
+        {
+            "date_col": ["2005-01-01", "2010-01-02", "2023-01-03"],
+            "email": [
+                "john.malkovich@bing.com",
+                "hannah.montana@disney.com",
+                "ilovedogs33@dyvenia.com",
+            ],
+            "name": ["John", "Hannah", "Patryk"],
+            "last_name": ["Malkovich", "Montana", "Dyvenian"],
+            "active": [1, 0, 0],
+        }
+    )
+
+    expected_output = {
+        "date_col": {0: "2005-01-01", 1: "2010-01-02", 2: "2023-01-03"},
+        "email": {
+            0: hash("john.malkovich@bing.com"),
+            1: hash("hannah.montana@disney.com"),
+            2: "ilovedogs33@dyvenia.com",
+        },
+        "name": {0: hash("John"), 1: hash("Hannah"), 2: "Patryk"},
+        "last_name": {0: hash("Malkovich"), 1: hash("Montana"), 2: "Dyvenian"},
+        "active": {0: 1, 1: 0, 2: 0},
+    }
+
+    output = anonymize_df.run(
+        data,
+        ["email", "name", "last_name"],
+        method="hash",
+        date_column="date_col",
+        days=2 * 365,
+    ).to_dict()
+    assert output == expected_output
+
+
+def test_anonymize_df_various_date_formats():
```

Comment:
You can try to add date formats:
`'20230101'` and `'2023-01-01T20:17:46.384Z'` or without 'Z'